pkg/util/gce/gce.go

```diff
@@ -62,6 +62,15 @@ func GetHostAlias() (string, error) {
 	return fmt.Sprintf("%s.%s", instanceName, projectID), nil
 }
 
+// GetClusterName returns the name of the cluster containing the current GCE instance
+func GetClusterName() (string, error) {
+	clusterName, err := getResponse(metadataURL + "/instance/attributes/cluster-name")
```

Comment:
We have some issues with AKS replying `200` with a HTML 404 content, is it safe to do this @arbll ?

Reply:
It will probably trigger the issue but I think it's fine to merge as is. We'll get this fixed on azure side or merge a mitigation before 6.11 in any case.